services/web/server/src/simcore_service_webserver/meta_version_control.py

```diff
@@ -0,0 +1,173 @@
+""" Access to the to version_control add-on
+
+"""
+
+import logging
+from types import SimpleNamespace
+from typing import List, Optional
+
+from aiopg.sa.result import RowProxy
+from models_library.projects import ProjectIDStr
+
+from .projects.project_models import ProjectDict
+from .version_control_changes import compute_workbench_checksum, eval_wcopy_project_id
+from .version_control_db import VersionControlRepository
+from .version_control_errors import UserUndefined
+from .version_control_models import CommitID, TagProxy
+from .version_control_tags import compose_wcopy_project_tag_name
+
+log = logging.getLogger(__name__)
+
+
+class VersionControlForMetaModeling(VersionControlRepository):
+
+    # TODO: eval inheritace vs composition?
+
+    async def get_wcopy_project_id(
+        self, repo_id: int, commit_id: Optional[int] = None
+    ) -> ProjectIDStr:
+        async with self.engine.acquire() as conn:
+            if commit_id is None:
+                commit = await self._get_HEAD_commit(repo_id, conn)
+                assert commit  # nosec
+                commit_id = commit.id
+                assert commit_id
+
+            return await self._fetch_wcopy_project_id(repo_id, commit_id, conn)
+
+    async def get_wcopy_project(self, repo_id: int, commit_id: int) -> ProjectDict:
+        async with self.engine.acquire() as conn:
+            project_id = await self._fetch_wcopy_project_id(repo_id, commit_id, conn)
+            project = await self.ProjectsOrm(conn).set_filter(uuid=project_id).fetch()
+            assert project  # nosec
+            return dict(project.items())
+
+    async def get_project(self, project_id: ProjectIDStr) -> ProjectDict:
+        async with self.engine.acquire() as conn:
+            if self.user_id is None:
+                # TODO: add message
+                raise UserUndefined
```

Comment:
the TODO is obsolete.

in python [both](https://docs.python.org/3.8/tutorial/errors.html#raising-exceptions), raising either an exception class or an instance, are valid. 

In this case, I use a technique i learned from pydantic library. If you look at ``UserDefined`` exception class, it is based on a template message that is filled with parameters. Since this exception has no parameters, there is no need to create an instance to raise it.

In any case, I will add it.